README.md

```diff
@@ -636,6 +653,136 @@ When defining an Audience, users and groups can be excluded from the audience. T
+
+``` C#
+public class Variant
+{
+    /// <summary>
+    /// The name of the variant.
+    /// </summary>
+    public string Name { get; set; }
+
+    /// <summary>
+    /// The configuration of the variant.
+    /// </summary>
+    public IConfigurationSection Configuration { get; set; }
+}
+```
+
+### Getting a Feature's Variant
+
+Variants expand on the basic flow of feature management and enable getting a feature's variant to later perform actions based on the name or configuration. This is done through the `IVariantFeatureManager`'s `GetVariantAsync` method.
+
+``` C#
+…
+IVariantFeatureManager featureManager;
+…
+Variant variant = await featureManager.GetVariantAsync(nameof(MyFeatureFlags.FeatureU));
+
+IConfigurationSection variantConfiguration = variant.Configuration;
+
+// Do something with the resulting variant and its configuration
+```
+
+### Setting a Variant's Configuration
+
+For each of the variants in the `Variants` property of a feature, there is a specified configuration. This can be set using either the `ConfigurationReference` or `ConfigurationValue` properties. `ConfigurationReference` is a string path that references a section of the current configuration that contains the feature flag declaration. `ConfigurationValue` is an inline configuration. If both are specified, `ConfigurationValue` is used.
+
+```
+"Variants": [
+    { 
+        "Name": "Big", 
+        "ConfigurationReference": "ShoppingCart:Big" 
+    },  
+    { 
+        "Name": "Small", 
+        "ConfigurationValue": {
+            "Size": 300
+        }
+    } 
+]
+```
+
+### Assigning a Variant
+
+The process of deciding which variant to return for a feature is called assignment. Assignment is determined by the `Allocation` property of the feature. Allocation logic is similar to the [Microsoft.Targeting](./README.md#MicrosoftTargeting) feature filter, with some additional parameters to allow for more specific variant assignment.
+
+```
+"Allocation": { 
+    "DefaultWhenEnabled": "Small", 
+    "DefaultWhenDisabled": "Small",  
+    "User": [ 
+        { 
+            "Variant": "Big", 
+            "Users": [ 
+                "Marsha" 
+            ] 
+        } 
+    ], 
+    "Group": [ 
+        { 
+            "Variant": "Big", 
+            "Groups": [ 
+                "Ring1" 
+            ] 
+        } 
+    ],
+    "Percentile": [ 
+        { 
+            "Variant": "Big", 
+            "From": 0, 
+            "To": 10 
+        } 
+    ], 
+    "Seed": "13973240" 
+},
+"Variants": [
+    { 
+        "Name": "Big", 
+        "ConfigurationReference": "ShoppingCart:Big" 
+    },  
+    { 
+        "Name": "Small", 
+        "ConfigurationValue": {
+            "Size": 300

```

Comment:
So this is required to be a JSON? Can I have more than one key value?

Reply:
Maybe I can explain that better, but it can be either a JSON or just a string after `ConfigurationValue` and it will return the configuration section starting with `ConfigurationValue` as the key. From there, you can index it like a normal configuration section. You can add any number of values or any structure of nested JSONs if you want.

Reply:
Can it be a number or a boolean?

Reply:
Yes, anything that could normally be in a configuration file like this, should I mention each of those?

Reply:
Yes, I think so. As an example, I think we should show a primitive value rather than a more complex JSON.